arch/arc/core/arc_smp.c

```diff
@@ -66,17 +66,41 @@ void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 	}
 }
 
+#ifdef CONFIG_SMP
+static void arc_connect_debug_mask_update(int cpu_num)
+{
+	uint32_t core_mask = 1 << cpu_num;
+
+	core_mask |= z_arc_connect_debug_select_read();
+	z_arc_connect_debug_select_set(core_mask);
+	/* Debugger halt cores at conditions */
+	z_arc_connect_debug_mask_set(core_mask,	(ARC_CONNECT_CMD_DEBUG_MASK_SH
+		| ARC_CONNECT_CMD_DEBUG_MASK_BH | ARC_CONNECT_CMD_DEBUG_MASK_AH
+		| ARC_CONNECT_CMD_DEBUG_MASK_H));
+}
+#endif
+
 /* the C entry of slave cores */
 void z_arc_slave_start(int cpu_num)
 {
 	arch_cpustart_t fn;
 
 #ifdef CONFIG_SMP
+	struct arc_connect_bcr bcr;
+
+	bcr.val = z_arc_v2_aux_reg_read(_ARC_V2_CONNECT_BCR);
+
+	if (bcr.dbg) {
+	/* configure inter-core debug unit if available */
```

Comment:
Either move this comment above `if` or add another identation level, i.e. move right to align with the next line `arc_connect_debug_mask_update(cpu_num);`.

Reply:
Fixed.